### ISSUE TITLE
[bug] `no-unnecessary-use-ref`: False positive when using ref to track whether/how often effect was executed

### ISSUE THREAD

**Issue by ulrichstark** [Rel1cx/eslint-react#1417]
### Describe the bug

This lint rule might be overly strict on flagging refs if they are used to track whether the effect was already executed or in a potential scenario where you want to track how many times it has already executed. The following repro code is distilled down from a real world scenario, where we use the ref to track if we still need to scrolll down, because we only want to do this once at the beginning.

### Reproduction

```ts
import { useEffect, useRef } from "react";

type Message = any;

declare function getPositionToScrollTo(messages: Message[]): number;

export function Chat({ messages }: { messages: Message[] }) {
  const hasScrolledRef = useRef(false);

  useEffect(() => {
    if (hasScrolledRef.current) return;

    window.scrollTo({ top: getPositionToScrollTo(messages) });
    hasScrolledRef.current = true;
  }, [messages]);

  return null;
}
```

### Expected behavior

That it doesn't flag my ref

### Platform and versions

```Text
@eslint-react/eslint-plugin: 2.7.0
```

### Stack trace

```Text

```

### Additional context

Similar to #1406.

Potential workarounds that might be cleaner React code anyway:
1. Remove ref, use empty dependencies array for effect, but then need to suppress lint `exhaustive-deps`
```ts
useEffect(() => {
  if (hasScrolledRef.current) return;

  window.scrollTo({ top: getPositionToScrollTo(messages) });
  hasScrolledRef.current = true;
// eslint-disable-next-line react-hooks/exhaustive-deps
}, []);
```

2. Remove effect and just do this side effect of scrolling directly in the component code
```ts
if (!hasScrolledRef.current) {
  window.scrollTo({ top: getPositionToScrollTo(messages) });
  hasScrolledRef.current = true;
}
```

Seeing my false positives I encountered with this lint rule and potential other false positives by others or me in the future, I think this rule might be overly strict in general. It found some helpful true positives in our codebase, but I think this lint rule is hard to get right.

**Comment by Rel1cx** [comment]
Thank you for the report. We'll explore our options, but as a temporary measure, I've removed this rule from the presets in [2.7.1](https://github.com/Rel1cx/eslint-react/releases/tag/v2.7.1).
